snackbar/material/task.md

```diff
@@ -0,0 +1,69 @@
+
+### Hinweise
+
+- Die Namen und Preise der Gerichte können fest codiert werden.
+- Es gibt eine Funktion in [`strings.h`](https://manual.cs50.io/#strings.h), die für die `get_cost`-Funktion hier nützlich sein könnte. _Welche ist das?_
+
+### Details
+
+Deine Aufgabe ist es nun eine `struct` und zwei Funktionen zu implementieren:
+
+1. Die `struct` `menu_item`, welche die Gerichte und Preise repräsentiert.
+2. Die Funktion `main`, um die Nutzereingaben zu verarbeiten und den Gesamtpreis zu berechnen.
+3. Die Funktion `get_cost`, die den Preis eines Gerichts anhand des Namens ermittelt.
+
+### Denkfrage
+
+Warum ist ein Array von `struct`s hier besser geeignet als mehrere Arrays?
+
+## Teilaufgabe 2
+
+Die angegeben Preise in der vorherigen Teilaufgabe waren ohne Mehrwertsteuer angegeben. In dieser Teilaufgabe sollst du die Berechnung der Mehrwertsteuer in die Funktionalität integrieren. Die Mehrwertsteuer beträgt **19%** und wird mithilfe der Konstante `VAT_RATE` definiert.
+
+### Aufgabe
+
+Erweitere die Funktionalität des existierenden Programms, sodass der letztendlich ausgegebene Preis inklusive **19%** Mehrwertsteuer ist.
+
+### Hinweis
+
+- Definiere die Mehrwertsteuer als Konstante.
+- Achte darauf, dass der berechnete Preis korrekt gerundet wird, falls nötig.
+- Überlege, an welcher Stelle die Berechnung der Mehrwertsteuer am sinnvollsten ist.
+
+## Teilaufgabe 3
+
+In dieser Teilaufgabe sollst du die `struct` und die Funktionen erweitern, um die Verfügbarkeit von Gerichten zu verwalten und zu überprüfen, ob die angeforderte Menge eines Gerichts bestellt werden kann.
+
+Erweitere die `struct` `menu_item`, indem du ein neues Feld `quantity` hinzufügst, das die verfügbare Anzahl jedes Gerichts speichert. Implementiere eine neue Funktion `find_item_by_name`, die eine **lineare Suche** durchführt, um ein Gericht anhand seines Namens zu finden, und den Index des entsprechenden Gerichts zurückgibt. Kann das angegebene Gericht nicht gefunden werden, so kann `-1` zurückgegeben werden. Refactore anschließend die Funktion `get_cost`, sodass sie den _Index_ des Gerichts von `find_item_by_name` erhält und die Menge überprüft. Berücksichtige, dass `find_item_by_name` den Wert `-1` zurück gibt, wenn das Gericht nicht gefunden wurde.
```

Comment:
As this is a "more-advanced" task, I'm not sure if we should enforce "namings" here, i.e. let the students think about what to do i.e. if no item could be found or how to actually store the quantity.

Again I'd add a paragraph to differentiate the sub-sub-tasks a bit easier